Library/Homebrew/cask/tab.rb

```diff
@@ -0,0 +1,112 @@
+# typed: true
+# frozen_string_literal: true
+
+require "tab"
+
+module Cask
+  class Tab < ::AbstractTab
+    attr_accessor :uninstall_flight_blocks, :uninstall_artifacts
+
+    # Instantiates a {Tab} for a new installation of a cask.
+    def self.create(cask)
+      tab = super
+
+      tab.tabfile = cask.metadata_main_container_path/FILENAME
+      tab.uninstall_flight_blocks = cask.uninstall_flight_blocks?
+      tab.runtime_dependencies = Tab.runtime_deps_hash(cask, cask.depends_on)
+      tab.source["version"] = cask.version.to_s
+      tab.source["path"] = cask.sourcefile_path.to_s
+      tab.uninstall_artifacts = cask.artifacts_list(uninstall_only: true)
+
+      tab
+    end
+
+    # Returns a {Tab} for an already installed cask,
+    # or a fake one if the cask is not installed.
+    def self.for_cask(cask)
+      path = cask.metadata_main_container_path/FILENAME
+
+      return from_file(path) if path.exist?
+
+      tab = empty
+      tab.source = {
+        "path"         => cask.sourcefile_path.to_s,
+        "tap"          => cask.tap&.name,
+        "tap_git_head" => (cask.tap&.installed? ? cask.tap.git_head : nil),
```

Comment:
```suggestion
        "tap_git_head" => (cask.tap.git_head if cask.tap&.installed?),
```
given this pattern is repeated 3+ times: feels like it might be nice to have a method that is either on `Tap` or passed a `Tap` to do this.